pkg/kubernetes/deployment/appsv1/v1alpha1/deployment.go

```diff
@@ -100,25 +105,248 @@ func (b *Builder) WithName(name string) *Builder {
+	}
+
+	for key, value := range annotations {
+		b.deployment.Object.Annotations[key] = value
+	}
+	return b
+}
+
+// WithAnnotationsNew resets existing annotaions if any with
+// ones that are provided here
+func (b *Builder) WithAnnotationsNew(annotations map[string]string) *Builder {
+	if len(annotations) == 0 {
+		b.errors = append(
+			b.errors,
+			errors.New("failed to build deployment object: no new annotations"),
+		)
+		return b
+	}
+
+	// copy of original map
+	newannotations := map[string]string{}
+	for key, value := range annotations {
+		newannotations[key] = value
+	}
+
+	// override
+	b.deployment.Object.Annotations = newannotations
+	return b
+}
+
+// WithLabels merges existing labels if any
+// with the ones that are provided here
+func (b *Builder) WithLabels(labels map[string]string) *Builder {
+	if len(labels) == 0 {
+		b.errors = append(
+			b.errors,
+			errors.New("failed to build deployment object: missing labels"),
+		)
+		return b
+	}
+
+	if b.deployment.Object.Labels == nil {
+		return b.WithLabelsNew(labels)
+	}
+
+	for key, value := range labels {
+		b.deployment.Object.Labels[key] = value
+	}
+	return b
+}
+
+// WithLabelsNew resets existing labels if any with
+// ones that are provided here
+func (b *Builder) WithLabelsNew(labels map[string]string) *Builder {
+	if len(labels) == 0 {
+		b.errors = append(
+			b.errors,
+			errors.New("failed to build deployment object: no new labels"),
+		)
+		return b
+	}
+
+	// copy of original map
+	newlbls := map[string]string{}
+	for key, value := range labels {
+		newlbls[key] = value
+	}
+
+	// override
+	b.deployment.Object.Labels = newlbls
+	return b
+}
+
+// WithMatchLabels merges existing matchlabels if any
+// with the ones that are provided here
+func (b *Builder) WithMatchLabels(matchlaebls map[string]string) *Builder {
+	if len(matchlaebls) == 0 {
+		b.errors = append(
+			b.errors,
+			errors.New("failed to build deployment object: missing matchlabels"),
+		)
+		return b
+	}
+
+	if b.deployment.Object.Spec.Selector.MatchLabels == nil {
+		return b.WithMatchLabelsNew(matchlaebls)
+	}
+
+	for key, value := range matchlaebls {
+		b.deployment.Object.Spec.Selector.MatchLabels[key] = value
+	}
+	return b
+}
+
+// WithMatchLabelsNew resets existing matchlabels if any with
+// ones that are provided here
+func (b *Builder) WithMatchLabelsNew(matchlaebls map[string]string) *Builder {
+	if len(matchlaebls) == 0 {
+		b.errors = append(
+			b.errors,
+			errors.New("failed to build deployment object: no new matchlabels"),
+		)
+		return b
+	}
+
+	// copy of original map
+	newmatchlabels := map[string]string{}
+	for key, value := range matchlaebls {
+		newmatchlabels[key] = value
+	}
+
+	// override
+	b.deployment.Object.Spec.Selector.MatchLabels = newmatchlabels
+	return b
+}
+
+// WithReplicas sets the replica field of deployment
+func (b *Builder) WithReplicas(replicas *int32) *Builder {
+
+	newreplicas := *replicas
+
+	if newreplicas <= 0 {
```

Comment:
a check for replicas being nil also would be good before even taken the value out of replicas..
error message also need to be fixed.. currently if I'm not wrong, its printing pointer

Reply:
Done

Reply:
0 is a valid count IMO